src/responder/secrets/local.c

```diff
@@ -372,14 +372,43 @@ int local_db_delete(TALLOC_CTX *mem_ctx,
                     struct local_context *lctx,
                     const char *req_path)
 {
+    TALLOC_CTX *tmp_ctx;
     struct ldb_dn *dn;
+    static const char *attrs[] = { NULL };
+    struct ldb_result *res;
     int ret;
 
+    tmp_ctx = talloc_new(mem_ctx);
+    if (!tmp_ctx) return ENOMEM;
+
     ret = local_db_dn(mem_ctx, lctx->ldb, req_path, &dn);
-    if (ret != EOK) return ret;
+    if (ret != EOK) goto done;
+
+    ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_BASE,
+                    attrs, LOCAL_CONTAINER_FILTER);
+    if (ret != EOK) goto done;
+
+    if (res->count == 1) {
+        ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_ONELEVEL,
+                         attrs, NULL);
+        if (ret != EOK) goto done;
+
+        if (res->count > 0) {
+            ret = EINVAL;
```

Comment:
On (09/09/16 13:12), Simo Sorce wrote:
>>      ret = local_db_dn(mem_ctx, lctx->ldb, req_path, &dn);
>> -    if (ret != EOK) return ret;
>> +    if (ret != EOK) goto done;
>> +
>> +    ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_BASE,
>> +                    attrs, LOCAL_CONTAINER_FILTER);
>> +    if (ret != EOK) goto done;
>> +
>> +    if (res->count == 1) {
>> +        ret = ldb_search(lctx->ldb, tmp_ctx, &res, dn, LDB_SCOPE_ONELEVEL,
>> +                         attrs, NULL);
>> +        if (ret != EOK) goto done;
>> +
>> +        if (res->count > 0) {
>> +            ret = EINVAL;
>
>open a ticket for that, it is a separte change
>
We(sssd) do not require ticket for any trivial change.
I let Fabiano decide wheter he wants to do it in this patch set
or in separate PR. Maybe he does not want to do it at all.

LS